src/components/Input/index.tsx

```diff
@@ -254,6 +237,70 @@ const InputMenuItemContainer = styled.li<InputMenuItemProps>`
   }
 `
 
+export const InputMenu = styled.div<React.HTMLAttributes<HTMLDivElement>>`
+  text-align: left;
+  position: absolute;
+  left: 0;
+  right: 0;
+  top: calc(100% + ${space[4]});
+  border-radius: ${radius.md};
+  background-color: ${color.nova};
+  box-shadow: ${shadow.strong};
+  overflow: hidden;
+  z-index: 1;
+
+  [data-theme='dark'] & {
+    border: 1px solid ${color.spaceLightest};
+  }
+`
+
+export interface InputMenuProps extends React.HTMLAttributes<HTMLDivElement> {
+  isOutsideViewport?: boolean
+}
+
+export interface StyledSelectMenuProps extends InputMenuProps {
+  heightOfMenu: number
+}
+
+const StyledSelectMenu = styled(InputMenu)<StyledSelectMenuProps>`
+  top: ${props =>
+    props.isOutsideViewport
+      ? `-${props.heightOfMenu + 4}px`
+      : 'calc(100% + 4px)'};
+`
+
+export const SelectMenu = ({ children, ...props }: InputMenuProps) => {
+  const [height, setHeight] = useState(0)
+  const [isOutsideViewport, setIsOutsideViewport] = useState(false)
+  const ref = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    setHeight(ref?.current?.clientHeight || 0)
+  }, [setHeight])
+
+  useEffect(() => {
+    const bottom = ref?.current?.getBoundingClientRect()?.bottom || 0
+    const viewportHeight = window?.innerHeight || 0
+
+    if (bottom > viewportHeight) {
+      setIsOutsideViewport(true)
```

Comment:
Shouldn't this be updated when scrolling? When a select is in a form at the bottom of a page it will go up. But the form goes further after scrolling, it should go down again.

Reply:
I don't see this as a problem at the moment. Bit of an edge case if users are trying to select something & then trying to scroll to the bottom. Imo it would also look weird if it changes if you focussing the input & you're scrolling.